tensorflow_addons/image/transform_ops.py

```diff
@@ -291,3 +292,55 @@ def _image_projective_transform_grad(op, grad):
         output_shape=tf.shape(image_or_images)[1:3],
         interpolation=interpolation)
     return [output, None, None]
+
+
+@tf.function
+def rotate(images, angles, interpolation="NEAREST", name=None):
+    """Rotate image(s) counterclockwise by the passed angle(s) in radians.
+
+    Args:
+      images: A tensor of shape (num_images, num_rows, num_columns, num_channels)
+        (NHWC), (num_rows, num_columns, num_channels) (HWC), or
+        (num_rows, num_columns) (HW). The rank must be statically known (the
+        shape is not `TensorShape(None)`.
+      angles: A scalar angle to rotate all images by, or (if images has rank 4)
+        a vector of length num_images, with an angle for each image in the batch.
+      interpolation: Interpolation mode. Supported values: "NEAREST", "BILINEAR".
+      name: The name of the op.
+
+    Returns:
+      Image(s) with the same type and shape as `images`, rotated by the given
+      angle(s). Empty space due to the rotation will be filled with zeros.
+
+    Raises:
+      TypeError: If `image` is an invalid type.
+    """
+    with ops.name_scope(name, "rotate"):
+        image_or_images = ops.convert_to_tensor(images)
+        if image_or_images.dtype.base_dtype not in _IMAGE_DTYPES:
+            raise TypeError("Invalid dtype %s." % image_or_images.dtype)
+        elif image_or_images.get_shape().ndims is None:
```

Comment:
How about use `if`, rather than `eilf` here?

Reply:
Do we really want to make non-code changes?  This was ported as-is from the other repo, so I feel it might be out of scope to make non-lint related changes.

Reply:
Because addons is designed for tf 2.0, in fact it seems not possible to migrate codes without any change. And also we need to maintain those codes as well, so it might make sense to make codes better. (We might also rewrite whole codes / add more test cases, in consider of out-of-date modules of tf.contrib)

Reply:
What do you think, Kyle?

Reply:
My vote is to keep it as-is under the idea that we should aim for minimal diffs whenever it's reasonable.

Reply:
I don't think putting `elif` makes sense here. Those conditions are different (checking dtype, checking shape).

Reply:
done